src/inc/regdisp.h

```diff
@@ -396,7 +396,7 @@ inline void FillRegDisplay(const PREGDISPLAY pRD, PT_CONTEXT pctx, PT_CONTEXT pC
 #elif defined(_TARGET_X86_) // _TARGET_ARM_
     for (int i = 0; i < 7; i++)
     {
-        *(&pRD->ctxPtrsOne.Esi + i) = (&pctx->Esi + i);
+        *(&pRD->ctxPtrsOne.Esi + i) = (&pRD->pCurrentContext->Esi + i);
```

Comment:
#9820 revises ``InitializeCurrentContextForCrawlFrame`` to update ``pRD->pCurrentContext`` using ``CurrentNonVolatileContextRecord`` saved in the dispatcher context, but ``pRD->pCurrentContextPointers`` points to ``pctx`` here, and thus it does not update ``*(pRD->pCurrentContextPointers->Ebp)``. This change is required to update ``*(pRD->pCurrentContextPointers->Ebp)``.

Internal evaluation shows that this does not introduce any regression, but we are currently not sure about the correctness of this change. 

Could you let us know why there are two distinct ``pContext`` and ``pCurrentContext``? As I understand, we may take another approach that revises ``InitializeCurrentContextForCrawlFrame`` to update ``pContext`` in order to fix this issue, but we are not sure about which approach is better.